java-checks/src/main/java/org/sonar/java/checks/helpers/RegexTreeHelper.java

```diff
@@ -253,4 +253,28 @@ public static boolean isEndBoundary(AutomatonState state) {
     }
   }
 
+  public static boolean onlyMatchesEmptySuffix(AutomatonState start) {
+    return onlyMatchesEmptySuffix(start, new HashSet<>());
+  }
+
+  private static boolean onlyMatchesEmptySuffix(AutomatonState start, Set<AutomatonState> visited) {
+    if (start instanceof FinalState || visited.contains(start)) {
+      return true;
+    }
+    visited.add(start);
+    if (start instanceof LookAroundTree) {
+      return onlyMatchesEmptySuffix(start.continuation());
```

Comment:
Why not `return onlyMatchesEmptySuffix(start.continuation(), visited);`?

Reply:
That was a mistake, good catch.